python/ray/data/_internal/stats.py

```diff
@@ -141,6 +153,46 @@ def __init__(self, max_stats=1000):
         self.max_stats = max_stats
         self.fifo_queue = []
 
+        # Ray Data dashboard metrics
+        # Everything is a gauge because we need to reset all of
+        # a dataset's metrics to 0 after each finishes execution.
+        tags_keys = ("dataset",)
+        self.bytes_spilled = Gauge(
+            DataMetric.BYTES_SPILLED.value,
+            description="Bytes spilled by dataset operators",
+            tag_keys=tags_keys,
+        )
+        self.bytes_allocated = Gauge(
+            DataMetric.BYTES_ALLOCATED.value,
+            description="Bytes allocated by dataset operators",
+            tag_keys=tags_keys,
+        )
+        self.bytes_freed = Gauge(
+            DataMetric.BYTES_FREED.value,
+            description="Bytes freed by dataset operators",
+            tag_keys=tags_keys,
+        )
+        self.bytes_current = Gauge(
+            DataMetric.BYTES_CURRENT.value,
+            description="Bytes currently in memory store used by dataset operators",
+            tag_keys=tags_keys,
+        )
+        self.cpu_usage = Gauge(
+            DataMetric.CPU_USAGE.value,
+            description="CPU cores used by dataset operators",
+            tag_keys=tags_keys,
+        )
+        self.gpu_usage = Gauge(
+            DataMetric.GPU_USAGE.value,
+            description="GPU cores used by dataset operators",
```

Comment:
```suggestion
            description="GPUs allocated to dataset operators",
```
Changed wording because "used" might make people think that this is the physical utilization.